src/application/use-cases/newsService/processNewsContent/processNewsContent.ts

```
@@ -0,0 +1,167 @@
+      result[key] = replaceImageSources(value as JsonValue, replacementMap);
+    } else {
+      result[key] = value;
+    }
+  }
+
+  return result as JsonValue;
+};
+
+export type LocalizedContent = {
+  uk: unknown;
+  en: unknown;
+};
+
+export type NewsContentInput = {
+  content: LocalizedContent;
+  description?: LocalizedContent;
+  coverImage?: {
+    src: string;
+    alt?: LocalizedContent;
+    caption?: LocalizedContent;
+    isTmp?: boolean;
+  };
+};
+
+const extractAllImageSources = (input: NewsContentInput): string[] => {
+  const allImages = new Set<string>();
+
+  if (input.content.uk) {
+    extractImageSrcs(input.content.uk as JsonValue).forEach((src) => allImages.add(src));
+  }
+
+  if (input.content.en) {
+    extractImageSrcs(input.content.en as JsonValue).forEach((src) => allImages.add(src));
+  }
+
+  if (input.description?.uk) {
+    extractImageSrcs(input.description.uk as JsonValue).forEach((src) => allImages.add(src));
+  }
+
+  if (input.description?.en) {
+    extractImageSrcs(input.description.en as JsonValue).forEach((src) => allImages.add(src));
+  }
+
+  if (input.coverImage?.src && input.coverImage?.isTmp === true) {
+    allImages.add(input.coverImage.src);
+  }
+
+  return Array.from(allImages);
+};
+
+const processImageUpload = async (
+  oldSrc: string,
+  uploadService: ReturnType<typeof blobStorageService>,
+  replacementMap: Map<string, string>
+): Promise<void> => {
+  if (!shouldProcessImageSource(oldSrc)) {
+    return;
+  }
+
+  try {
+    await uploadService.copyBlobsToNewFolder('tmp', 'photos', [oldSrc]);
+
+    const newUrl = uploadService.constructBlobUrl('photos', oldSrc);
+    replacementMap.set(oldSrc, newUrl);
+
+    try {
+      await uploadService.deleteFile('tmp', oldSrc);
+    } catch (deleteError) {
+      console.error(`Failed to delete tmp blob ${oldSrc}:`, deleteError);
+    }
+  } catch (error) {
+    console.error(`Failed to process image ${oldSrc}:`, error);
```

Avoid silent failure, as it leads to data inconsistency. Use `throw new Error()`.